spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraTableScan.scala

```diff
@@ -18,8 +25,11 @@ case class CassandraTableScan(
 
   private def inputRdd = {
     logInfo(s"attributes : ${attributes.map(_.name).mkString(",")}")
-    //TODO: cluster level CassandraConnector, read configuration settings
-    var rdd = context.sparkContext.cassandraTable[CassandraSQLRow](relation.keyspaceName, relation.tableName)
+    val readConf = ReadConf.fromSparkConf(context.getReadConf(relation.keyspaceName, relation.tableName, relation.cluster))
+    var rdd = context.sparkContext.cassandraTable[CassandraSQLRow](
+      relation.keyspaceName, relation.tableName, readConf)(
+      CassandraConnector(context.getCassandraConnConf(relation.cluster)), ClassTag(CassandraSQLRow.getClass),
```

Comment:
Here also you do not need to call explicitly `ClassTag` - try `implicitly[ClassTag[CassandraSQLRow]]`